app/src/main/java/com/gruppo4/SMSApp/MainActivity.java

```diff
@@ -0,0 +1,140 @@
+package com.gruppo4.SMSApp;
+
+/**
+ * @author Francesco Bau'
+ */
+
+import com.gruppo4.audioUtility.AudioUtilityManager;
+
+import android.content.Context;
+import android.media.AudioManager;
+import android.os.Bundle;
+import android.os.Vibrator;
+import android.view.View;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+
+public class MainActivity extends AppCompatActivity {
+
+
+    // Default app volume, expressed in percentage
+    private static final int DEFAULT_APP_VOLUME = 50;
+    // The selected Stream.
+    private static AudioUtilityManager.AUMStream stream;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
```

Comment:
Instead of adding all the onClick event handlers in the java file (resulting in a huge `onCreate(...)` method), consider using the `android:onClick` XML attribute. You just need to create one method for each button and linking it with the corresponding UI component event in the XML layout file using the attribute written before.
By doing this you have:

- More readable code
- The possibility to name methods accordingly, for example, instead of having all the handlers named `onClick(...)`, as it is now, you can name them like this: `alarmButtonOnClick(..)`, `musicButtonOnClick(...)`, ...
- The possibility to add Javadoc specifics at the beginning of each onClick event handler method, making it even more informative

Reply:
Thank you for the tip...
I've never tried that way.

Reply:
Yes, it is. If you put the method name in the field the IDE will automatically add the `android:onClick` attribute in the XML file with the given name (note that the IDE does not create the method, you need to create it manually in the activity java file).

Reply:
I see... I will improve it next time. I don't want to do a drastic change right now, or I could make things worse than before.
I've tried to see how to make it, but I don't think I've understold enough.
But thank you again, I'll experiment in that way, like you told me.